__test__/rest/user.test.js

```diff
@@ -0,0 +1,81 @@
+const request = require('supertest');
+const app = require('../../app');
+const mongoose = require('mongoose');
+const {settings, dbURI} = require('../../config/db');
+const User = require('../../models/User');
+const faker = require('faker');
+
+
+describe('some initial tests for user', () => {
+    //global variables that we can use in all tests
+    let randomUser, registerResponse, connection;
+
+    beforeAll(async done =>{
+        //waiting for connection with mongoDB database
+        connection = await mongoose.connect(dbURI, settings);
+        done()
+    });
+    test('should create new user', async () => {
+        //create user with random properties.
+        const userToRegister = {
+            firstName: faker.name.firstName(),
+            lastName: faker.name.lastName(),
+            email: faker.internet.email(),
+            password: faker.internet.password()
+        };
+
+        const response = await request(app).post('/api/users/register')
+            .type('form')
+            .send({...userToRegister})
+            .set('Accept', 'application/json');
+        expect(response.body.msg).toContain('registered');
+        expect(response.body.token).toContain('Bearer');
+
+        const user = await User.findOne({email: userToRegister.email});
```

Comment:
Can't you use destructuring in that case?

```const { email, firstName } = userToRegister```

Reply:
I don't know what is best practice here. If we creating object 
```
const userToRegister = {
            firstName: faker.name.firstName(),
            lastName: faker.name.lastName(),
            email: faker.internet.email(),
            password: faker.internet.password()
        };
```
only to descructure this object later mb we could just create 4 properties from start like:

```
// user data to register
const firstName: faker.name.firstName(),
const lastName: faker.name.
const email: faker.internet.email(),
const password: faker.internet.password()
```
What do you think about that? 
I'm sure I'v could do:
`const { body } = await request(app).post('/api/users/register')`
`expect(body.msg).toContain('registered');`
`expect(body.token).toContain('Bearer');`
or even 
`const { body: {msg, token} } = await request(app).post('/api/users/register')`

Reply:
`const { body } = await request(app).post('/api/users/register')`

I prefer that way. Nested destructiring doesn't look so great